synapse/storage/_base.py

```diff
@@ -547,11 +547,19 @@ def _simple_upsert_txn(self, txn, table, keyvalues, values, insertion_values={},
         if lock:
             self.database_engine.lock_table(txn, table)
 
+        def _getwhere(val):
+            # If the value we're passing in is None (aka NULL), we need to use
+            # IS, not =, as NULL = NULL equals NULL (False).
+            if values.get(val, keyvalues.get(val)) is None:
```

Comment:
why are we checking `values` here?

(and why `keyvalues.get` rather than `keyvalues[]` ?)

Reply:
because i'm silly